gtfs-realtime/spec/en/reference.md

```diff
@@ -393,15 +406,18 @@ Identification information for the vehicle performing the trip.
 
+At least one specifier must be given - all fields in an EntitySelector cannot be empty.
+
 #### Fields
 
 | _**Field Name**_ | _**Type**_ | _**Required**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|----------------|-------------------|-------------------|
-| **agency_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | At least one specifier must be given - all fields in an EntitySelector cannot be empty.
-| **route_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | At least one specifier must be given - all fields in an EntitySelector cannot be empty.
-| **route_type** | [int32](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | At least one specifier must be given - all fields in an EntitySelector cannot be empty.
-| **trip** | [TripDescriptor](#message-tripdescriptor) | Conditionally required | One | At least one specifier must be given - all fields in an EntitySelector cannot be empty.
-| **stop_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | At least one specifier must be given - all fields in an EntitySelector cannot be empty.
+| **agency_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | The agency_id from the GTFS feed that this selector refers to.
+| **route_id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | The route_id from the GTFS that this selector refers to. If direction_id is provided, route_id must also be provided.
+| **route_type** | [int32](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | The route_type from the GTFS that this selector refers to.
+| **direction_id** | [uint32](https://developers.google.com/protocol-buffers/docs/proto#scalar) | Conditionally required | One | The direction_id from the GTFS feed trips.txt file, used to select all trips in one direction for a route, specified by route_id. If direction_id is provided, route_id must also be provided. <br>**Caution:** this field is still **experimental**, and subject to change. It may be formally adopted in the future.<br>|
+| **trip** | [TripDescriptor](#message-tripdescriptor) | Conditionally required | One | The trip instance from the GTFS that this selector refers to. This TripDescriptor must resolve to a single trip instance in the GTFS data (e.g., a producer cannot provide only a trip_id for exact_times=0 trips). If the ScheduleRelationship field is populated within this TripDescriptor it will be ignored by consumers when attempting to identify the GTFS trip.
```

Comment:
(e.g., a producer cannot provide only a trip_id for **exact_times=0** trips)

I think this is true for all trips defined in frequencies.txt, not only exact_times=0. A start_time must always be provided (as stated above)

Reply:
@tleboulenge Yes, that is correct, all trips defined in frequencies.txt must include a `start_time` to be valid.  The example I gave here wasn't intended to be exclusionary (i.e., wasn't intended to imply that `exact_times=1` trips don't require `start_time`) - if you think this should be clarified in the text let me know and we can restart the vote.

Reply:
I see my comment as just a detail, akin to a typo, wouldn't require restarting the vote for this. I'll leave it up to you to decide to go ahead as it is, or update that wording.